src/cascadia/WindowsTerminal/AppHost.cpp

```diff
@@ -188,54 +188,25 @@ void AppHost::_HandleCreateWindow(const HWND hwnd, RECT proposedRect, winrt::Ter
-        const short _currentHeight = Utils::ClampToShortMax(
+        const short islandHeight = Utils::ClampToShortMax(
             static_cast<long>(ceil(initialSize.Y)), 1);
 
-        // Create a RECT from our requested client size
-        auto nonClient = Viewport::FromDimensions({ _currentWidth,
-                                                    _currentHeight })
-                             .ToRect();
+        RECT islandFrame = {};
+        bool succeeded = AdjustWindowRectExForDpi(&islandFrame, WS_OVERLAPPEDWINDOW, false, 0, dpix);
+        // If we failed to get the correct window size for whatever reason, log
+        // the error and go on. We'll use whatever the control proposed as the
+        // size of our window, which will be at least close.
+        LOG_LAST_ERROR_IF(!succeeded);
 
-        // Get the size of a window we'd need to host that client rect. This will
-        // add the titlebar space.
         if (_useNonClientArea)
         {
-            // If we're in NC tabs mode, do the math ourselves. Get the margins
-            // we're using for the window - this will include the size of the
-            // titlebar content.
-            const auto pNcWindow = static_cast<NonClientIslandWindow*>(_window.get());
-            const MARGINS margins = pNcWindow->GetFrameMargins();
-            nonClient.left = 0;
-            nonClient.top = 0;
-            nonClient.right = margins.cxLeftWidth + nonClient.right + margins.cxRightWidth;
-            nonClient.bottom = margins.cyTopHeight + nonClient.bottom + margins.cyBottomHeight;
-        }
-        else
-        {
-            bool succeeded = AdjustWindowRectExForDpi(&nonClient, WS_OVERLAPPEDWINDOW, false, 0, dpix);
-            if (!succeeded)
-            {
-                // If we failed to get the correct window size for whatever reason, log
-                // the error and go on. We'll use whatever the control proposed as the
-                // size of our window, which will be at least close.
-                LOG_LAST_ERROR();
-                nonClient = Viewport::FromDimensions({ _currentWidth,
-                                                       _currentHeight })
-                                .ToRect();
-            }
-
-            // For client island scenario, there is an invisible border of 8 pixels.
-            // We need to remove this border to guarantee the left edge of the window
-            // coincides with the screen
-            const auto pCWindow = static_cast<IslandWindow*>(_window.get());
-            const RECT frame = pCWindow->GetFrameBorderMargins(dpix);
-            proposedRect.left += frame.left;
+            islandFrame.top = -NonClientIslandWindow::topBorderVisibleHeight;
         }
 
-        adjustedHeight = nonClient.bottom - nonClient.top;
-        adjustedWidth = nonClient.right - nonClient.left;
+        adjustedWidth = -islandFrame.left + islandWidth + islandFrame.right;

```

Comment:
I'd maybe add a comment on why we don't need to do anything for the `!_useNonClientArea` case here